src/resolver.rs

```diff
@@ -389,6 +416,10 @@ impl AnnotationMapImpl {
         self.type_hint_map.insert(s.get_id(), annotation);
     }
 
+    pub fn annotate_function(&mut self, s: &AstStatement, f: AstStatement) {
```

Comment:
i find this name too generic, it sounds like it would annotate a function-pou

what do you think about: 

/// annotates the given statement `s` with the call-statement `f` so codegen can generate
/// a hidden call `f` instead of generating `s`
pub fn annotate_hidden_function_call(...)